celer/PN_logreg.pyx

```diff
@@ -400,11 +400,12 @@ cpdef void do_line_search(
         floating[::1, :] X, floating[:] X_data,
         int[:] X_indices, int[:] X_indptr, int MAX_BACKTRACK_ITR,
         floating[:] y, floating[:] exp_Xw, floating[:] low_exp_Xw,
-        floating[:] aux, int[:] is_positive_label) nogil:
+        floating[:] aux, int[:] is_positive_label):
```

Comment:
I think not releasing the GIL here won't affect the performance as this function is originally single-threaded.